db/db_impl/db_impl.cc

```diff
@@ -1570,7 +1570,11 @@ Status DBImpl::ApplyWALToManifest(VersionEdit* synced_wals) {
 }
 
 Status DBImpl::LockWAL() {
-  log_write_mutex_.Lock();
+  {
+    InstrumentedMutexLock lock(&mutex_);
+    lock_wal_write_token_ = write_controller_.GetStopToken();
```

Comment:
Make sense. I originally assumed application will make sure all writes complete. Will look into adding this, otherwise, a future backup may race with some WAL writes.

Reply:
Done.